src/python/pants/engine/internals/selectors.py

```diff
@@ -26,11 +29,83 @@
 _SubjectType = TypeVar("_SubjectType")
 
 
+class GetParseError(ValueError):
+    def __init__(
+        self, explanation: str, *, get_args: Sequence[ast.expr], source_file_name: str
+    ) -> None:
+        def render_arg(expr: ast.expr) -> str:
+            if isinstance(expr, ast.Name):
+                return expr.id
+            if isinstance(expr, ast.Call):
+                # Check if it's a top-level function call.
+                if hasattr(expr.func, "id"):
+                    return f"{expr.func.id}()"  # type: ignore[attr-defined]
+                # Check if it's a method call.
+                if hasattr(expr.func, "attr") and hasattr(expr.func, "value"):
+                    return f"{expr.func.value.id}.{expr.func.attr}()"  # type: ignore[attr-defined]
+
+            # Fall back to the name of the ast node's class.
+            return str(type(expr))
```

Comment:
The original code used `type(expr).__name__`, but I think this made the error confusing because we would output `Str` rather than `<class '_ast.Str'>`.